### ISSUE TITLE
Lint/RedundantCopDisableDirective raises an exception when disabling an entire department

### ISSUE THREAD

**Issue by thomasnemer** [rubocop/rubocop#11145]
When using `#rubocop:disable Metrics` on a block of code and not all Metrics/* cops would have raised an offense if disabled individually instead of disabling the entire cop department, `Lint/RedundantCopDisableDirective` raises an exception telling `Couldn't find Metrics/CyclomaticComplexity in comment: # rubocop:todo Metrics`

This is true the cop should probably not be disabled if it would not have raised an offense, but maybe this should be reported as an offense, not raise an exception.

--------

## Expected behavior

I would have expected rubocop to tell me that  `Some/CopA`, `Some/CopB` and `Some/CopC` should explicitely be disabled instead of disabling the entire `Some` department

## Actual behavior

`Lint/RedundantCopDisableDirective raises an exception` raises an exception telling some specific cop is not found in a comment : 

```
An error occurred while Lint/RedundantCopDisableDirective cop was inspecting app/some_path/some_file_controller.rb.
Couldn't find Metrics/CyclomaticComplexity in comment: # rubocop:todo Metrics/
~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:256:in `cop_range'
~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:233:in `block in add_offense_for_some_cops'
~/.rbenv/versions/3.1.2/lib/ruby/3.1.0/set.rb:511:in `each_key'
~/.rbenv/versions/3.1.2/lib/ruby/3.1.0/set.rb:511:in `each'
~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:233:in `map'
~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:233:in `add_offense_for_some_cops'
~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:212:in `block in add_offenses'
~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:208:in `each'
~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:208:in `add_offenses'
~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:51:in `on_new_investigation'
```

## Steps to reproduce the problem

test.rb : 
```ruby
class Test
  # rubocop:todo Metrics
  def create
    bool_a = true
    bool_b = true
    bool_c = true
    bool_d = true
    bool_e = true
    if bool_a
      if bool_b
        if bool_c
          if bool_d
            if bool_e
              puts 'this line need to be long=========================================================================='
            end
          end
        end
      end
    end
  end
  # rubocop:enable Metrics
end
```
`bundle exec rubocop -d test.rb`

```
Scanning test.rb
An error occurred while Lint/RedundantCopDisableDirective cop was inspecting test.rb.
Couldn't find Metrics/CyclomaticComplexity in comment: # rubocop:todo Metrics
```

## RuboCop version
```
$ bundle exec rubocop -V
1.38.0 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 3.1.2) [x86_64-linux]
  - rubocop-minitest 0.22.2
  - rubocop-performance 1.15.0
  - rubocop-rails 2.16.1
  - rubocop-rake 0.6.0
```

**Comment by fatkodima** [comment]
I was not able to reproduce it. Can you verify this issue still persists using rubocop `master`? 
Your problem is probably fixed by https://github.com/rubocop/rubocop/pull/11210.

**Comment by thomasnemer** [comment]
Hi :wave:

Thanks for having a look. :bow: 

I confirm this issue is now fixed :)
